doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,152 @@
+# Audit Trail MVP
+
+NIH has required that DANDI Archive generate audit logs while leaving undefined
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action
```

Comment:
just a note here that the authorized user can be a backend task (i.e. the archive). for example when an asset summary is updated or checksum is computed.

Reply:
Is there a reason users (i.e., Dandiset owners) would want to know when these things are computed/updated?

Reply:
knowing that something has changed in the dandiset is important. as currently listed one should consider any modification a significant event (independent of agent). we can/should list examples of non-significant events if we currently think that a user wouldn't care, and we can discuss those to moved from one pile to another as needed. 

for the future, we will continue to add tasks that will make changes, some more important than others (e.g., generating a derived asset, altering metadata, etc.,.). hence i already want to start having a notion that an internal action is still an action that should be recorded.

Reply:
See de07b152c51ffc139a9a61f65057b6cf3bd64c0a.